core/src/mast/mod.rs

```diff
@@ -153,11 +153,10 @@ impl MastForest {
 
     /// Removes all decorators from this MAST forest.
     pub fn strip_decorators(&mut self) {
-        for node in self.nodes.iter_mut() {
-            node.remove_decorators();
-        }
+        // Clear all decorator storage
         self.decorators = IndexVec::new();
         self.op_decorator_storage = OpToDecoratorIds::new();
+        self.node_decorator_storage.clear();
     }
```

Comment:
Comment for the future: once decorators are stripped, we'll also need to run a "forest compaction" procedure because we may be able to eliminate some nodes which previously differed only by decorators. This may already be captured in one of the issues, but if not, let's note this.